hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java

```diff
@@ -2945,6 +2945,9 @@ private void processRpcRequest(RpcRequestHeaderProto header,
             }
           }
         } catch (IOException ioe) {
+          if (ioe instanceof RetriableException) {
```

Comment:
Consider putting a try catch blocked around receiveRequestState. This make it easier to see where the exception is coming from, instead of checking the type of exception later. So something like
```java
try {
    stateId = alignmentContext.receiveRequestState(header, getMaxIdleTime());
} catch (RetriableException re) {
    rpcMetrics.incrRcRejectedByObserverCalls();
    throw re;
}
```

Reply:
Sir  @simbadzina thanks help me review and suggestions.

I will update later PR.